sklearn/linear_model/bayes.py

```diff
@@ -224,12 +231,19 @@ def fit(self, X, y, sample_weight=None):
                                               XT_y, U, Vh, eigen_vals_,
                                               alpha_, lambda_)
             if self.compute_score:
-                # compute the log marginal likelihood
-                s = self._log_marginal_likelihood(n_samples, n_features,
-                                                  eigen_vals_,
-                                                  alpha_, lambda_,
-                                                  coef_, rmse_)
-                self.scores_.append(s)
+                # compute the objective function
+                L = self._compute_lower_bound(n_samples, n_features,
+                                              eigen_vals_, alpha_, lambda_,
+                                              coef_, rmse_, L_offset)
+                self.scores_.append(L)
+
+            # Check for convergence
```

Comment:
you permuted the check for convergence and the update of alpha and lambda because we do not need to do the update if converge = True, is that correct?

Reply:
Yes, it's correct, I think.
Before correction, the objective function value and the result coef_ conflicted.